### PR TITLE
Sort OS versions taking their rev into account if present

### PR DESCRIPTION
**Pull request by pimterry**
The test at the very bottom is the best place to look if you're just checking you're happy with the resulting order we'll be seeing.

Recommended logic doesn't change, that's still just "first in the list that's not prerelease".

### CODE REVIEW

File: lib/util/index.coffee [hunk 1]
```diff
@@ -39,6 +40,26 @@ exports.treatAsMissingDevice = (uuidOrId) ->
...
 
+exports.osVersionCompare = (versionA, versionB) ->
+	semverResult = semver.rcompare(versionA, versionB)
+	if semverResult != 0
+		return semverResult
+
+	revA = getRev(versionA)
+	revB = getRev(versionB)
+
+	return revB - revA
```
Comment (emirotin) [inline lib/util/index.coffee]: NIT: this coerces `null` to `0` which works but is slightly ugly, I'd just return `0` from `getRev`

File: lib/util/index.coffee [hunk 2]
```diff
@@ -39,6 +40,26 @@ exports.treatAsMissingDevice = (uuidOrId) ->
 		replacementErr.stack = err.stack
 		throw replacementErr
 
+exports.osVersionCompare = (versionA, versionB) ->
```
Comment (emirotin) [inline lib/util/index.coffee]: Also NIT: this is R-compare so I would rename the method to make it obvious the comparison is inversed here.

**Comment by lurch** [comment]
Hah, looks a bit like the code @jviotti is adding to Etcher in a recent PR ;-)
https://github.com/resin-io/etcher/pull/1183/files#diff-761eb090648cc858c72e655f2a6171bcR206

**Comment by lurch** [comment]
P.S. @pimterry Should this be connecting to #302 ?

File: lib/util/index.coffee (hunk 2 above)
Comment (pimterry) [inline lib/util/index.coffee]: Sure, done.

**Comment by pimterry** [comment]
@lurch yes, it should, thanks! Sort-of similar, although that's how ranges match to prerelease tags, and this ordering of build tags. It might be nice to share all this logic eventually though. Maybe in the medium term we should have a resin-semver module, which extends semver to take prerelease/build metadata into account in the way we'd like? Could be useful, especially if more of this emerges in future.

File: lib/util/index.coffee (hunk 1 above)
Comment (pimterry) [inline lib/util/index.coffee]: Sure, done.

**Pull request merged**